### PR TITLE
coco/dragon: add midi cartridge

### PR DESCRIPTION
**Pull request by tlindner**
Implement a midi device as a coco/dragon cartridge.

### CODE REVIEW

**Comment by rb6502** [comment]
This looks good as far as 6850 MIDI goes.  Have you tested it yourself?

File: src/devices/bus/coco/coco_midi.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    m_acia->write_txc(state);
+    m_acia->write_rxc(state);
+}
+
+WRITE_LINE_MEMBER(coco_dragon_midi_device::midi_in)
+{
+    // MIDI in signals is sent to both the 6850 and the MIDI thru output port
+    m_acia->write_rxd(state);
+    m_mdthru->write_txd(state);
+}
```
Comment (cuavas) [inline src/devices/bus/coco/coco_midi.cpp:125]: You can do all this in machine config, there’s no need for additional functions.

File: src/devices/bus/coco/coco_midi.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+***************************************************************************/
+
+#include "emu.h"
+#include "coco_midi.h"
+
+#include "machine/6850acia.h"
+#include "machine/clock.h"
+#include "bus/midi/midi.h"
+
+#define MC6850_TAG "mc6850"
```
Comment (cuavas) [inline src/devices/bus/coco/coco_midi.cpp:19]: It’s pointless having this macro when you use the finder to access it everywhere anyway.  The literal tag wouldn’t be repeated even without the macro.

File: src/devices/bus/coco/coco_midi.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+{
+    install_readwrite_handler(0xff74, 0xff75,
+            read8sm_delegate(m_acia, FUNC(acia6850_device::read)),
+            write8sm_delegate(m_acia, FUNC(acia6850_device::write)));
+}
+
+void coco_dragon_midi_device::device_reset()
+{
+    m_acia->reset();
+}
```
Comment (cuavas) [inline src/devices/bus/coco/coco_midi.cpp:112]: Reset propagates to child devices automatically.  You shouldn’t need to be doing this.

**Review by cuavas — CHANGES_REQUESTED** [review]
I’ve said it before, I’ll say it again – if a class is declared inside an anonymous namespace, it should be implemented inside the anonymous namespace.  Also, namespaces are not declarations and hence don’t need a terminating semicolon.

**Comment by tlindner** [comment]
@rb6502 I've tested midi out. I really should test midi in before this is committed. I will report back.
@cuavas I believe I've addresses all your comments.

File: src/devices/bus/coco/coco_midi.cpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		install_readwrite_handler(0xff74, 0xff75,
+				read8sm_delegate(m_acia, FUNC(acia6850_device::read)),
+				write8sm_delegate(m_acia, FUNC(acia6850_device::write)));
+	}
+
+	WRITE_LINE_MEMBER( coco_dragon_midi_device::write_acia_clock )
+	{
+		m_acia->write_txc(state);
+		m_acia->write_rxc(state);
+	}
```
Comment (cuavas) [inline src/devices/bus/coco/coco_midi.cpp:105]: You can do this in machine config as well.

**Comment by tlindner** [comment]
I see that I am not validating. Looking into it now.

**Pull request merged**